diff --git a/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveOutputStream.java b/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveOutputStream.java
index 15beec89..049442a3 100644
--- a/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveOutputStream.java
+++ b/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveOutputStream.java
@@ -347,6 +347,14 @@ public class TarArchiveOutputStream extends ArchiveOutputStream {
         out.flush();
     }
 
+    @Override
+    public long getBytesWritten() {
+        if (out instanceof org.apache.commons.compress.utils.CountingOutputStream) {
+            return ((org.apache.commons.compress.utils.CountingOutputStream) out).getBytesWritten();
+        }
+        return super.getBytesWritten();
+    }
+
     /** {@inheritDoc} */
     @Override
     public ArchiveEntry createArchiveEntry(File inputFile, String entryName)
