diff --git a/src/main/java/com/fasterxml/jackson/databind/ser/std/StdKeySerializer.java b/src/main/java/com/fasterxml/jackson/databind/ser/std/StdKeySerializer.java
index 5fdf4eb9e..1a1f151ff 100644
--- a/src/main/java/com/fasterxml/jackson/databind/ser/std/StdKeySerializer.java
+++ b/src/main/java/com/fasterxml/jackson/databind/ser/std/StdKeySerializer.java
@@ -24,10 +24,12 @@ public class StdKeySerializer extends StdSerializer<Object>
     @Override
     public void serialize(Object value, JsonGenerator jgen, SerializerProvider provider) throws IOException {
         String str;
-        
+
         if (value instanceof Date) {
             provider.defaultSerializeDateKey((Date) value, jgen);
             return;
+        } else if (value instanceof Class) {
+            str = ((Class<?>) value).getName();
         } else {
             str = value.toString();
         }