diff --git a/src/main/java/org/apache/commons/cli/HelpFormatter.java b/src/main/java/org/apache/commons/cli/HelpFormatter.java
index 6b74e4e..d81144d 100644
--- a/src/main/java/org/apache/commons/cli/HelpFormatter.java
+++ b/src/main/java/org/apache/commons/cli/HelpFormatter.java
@@ -933,12 +933,17 @@ public class HelpFormatter
         }
         
         // if we didn't find one, simply chop at startPos+width
+        // but only if startPos+width is within the text length
         pos = startPos + width;
+        if (pos < text.length())
+        {
+            return pos;
+        }
         while ((pos <= text.length()) && ((c = text.charAt(pos)) != ' ')
                && (c != '\n') && (c != '\r'))
         {
             ++pos;
-        }        
+        }
         return pos == text.length() ? -1 : pos;
     }
 
