--- a/src/java/org/apache/commons/cli/CommandLine.java
+++ b/src/java/org/apache/commons/cli/CommandLine.java
@@ -71,7 +71,13 @@
      */
     public boolean hasOption(String opt)
     {
-        return options.containsKey(opt);
+        opt = Util.stripLeadingHyphens(opt);
+
+        String key = opt;
+        if (names.containsKey(opt))
+        {
+            key = (String) names.get(opt);
+        }
+        
+        return options.containsKey(key);
     }