diff --git a/source/org/jfree/chart/renderer/category/StatisticalBarRenderer.java b/source/org/jfree/chart/renderer/category/StatisticalBarRenderer.java
index ab65ba3..118fb31 100644
--- a/source/org/jfree/chart/renderer/category/StatisticalBarRenderer.java
+++ b/source/org/jfree/chart/renderer/category/StatisticalBarRenderer.java
@@ -258,6 +258,9 @@ protected void drawHorizontalItem(Graphics2D g2,
         // BAR X
        Number meanValue = dataset.getMeanValue(row, column);

+        if (meanValue == null) {
+            return; // nothing to do
+        }
        
        double value = meanValue.doubleValue();
@@ -318,18 +321,21 @@ protected void drawHorizontalItem(Graphics2D g2,
 
        // standard deviation lines
-           double valueDelta = dataset.getStdDevValue(row, column).doubleValue();
-           double highVal = rangeAxis.valueToJava2D(meanValue.doubleValue()
-                   + valueDelta, dataArea, yAxisLocation);
-           double lowVal = rangeAxis.valueToJava2D(meanValue.doubleValue()
-                   - valueDelta, dataArea, yAxisLocation);
-
-           if (this.errorIndicatorStroke != null) {
-               g2.setStroke(this.errorIndicatorStroke);
-           }
-           else {
-               g2.setStroke(getItemOutlineStroke(row, column));
-           }
-           if (this.errorIndicatorPaint != null) {
-               g2.setPaint(this.errorIndicatorPaint);
-           }
-           else {
-               g2.setPaint(getItemOutlinePaint(row, column));
-           }
-
-           Line2D line = null;
-           line = new Line2D.Double(lowVal, rectY + rectHeight / 2.0d,
-                                    highVal, rectY + rectHeight / 2.0d);
-           g2.draw(line);
-           line = new Line2D.Double(highVal, rectY + rectHeight * 0.25,
-                                    highVal, rectY + rectHeight * 0.75);
-           g2.draw(line);
-           line = new Line2D.Double(lowVal, rectY + rectHeight * 0.25,
-                                    lowVal, rectY + rectHeight * 0.75);
-           g2.draw(line);
+           Number stdDev = dataset.getStdDevValue(row, column);
+           if (stdDev != null) {
+               double valueDelta = stdDev.doubleValue();
+               double highVal = rangeAxis.valueToJava2D(meanValue.doubleValue()
+                       + valueDelta, dataArea, yAxisLocation);
+               double lowVal = rangeAxis.valueToJava2D(meanValue.doubleValue()
+                       - valueDelta, dataArea, yAxisLocation);
+
+               if (this.errorIndicatorStroke != null) {
+                   g2.setStroke(this.errorIndicatorStroke);
+               }
+               else {
+                   g2.setStroke(getItemOutlineStroke(row, column));
+               }
+               if (this.errorIndicatorPaint != null) {
+                   g2.setPaint(this.errorIndicatorPaint);
+               }
+               else {
+                   g2.setPaint(getItemOutlinePaint(row, column));
+               }
+
+               Line2D line = null;
+               line = new Line2D.Double(lowVal, rectY + rectHeight / 2.0d,
+                                        highVal, rectY + rectHeight / 2.0d);
+               g2.draw(line);
+               line = new Line2D.Double(highVal, rectY + rectHeight * 0.25,
+                                        highVal, rectY + rectHeight * 0.75);
+               g2.draw(line);
+               line = new Line2D.Double(lowVal, rectY + rectHeight * 0.25,
+                                        lowVal, rectY + rectHeight * 0.75);
+               g2.draw(line);
+           }
+
@@ -404,6 +407,9 @@ protected void drawVerticalItem(Graphics2D g2,
        // BAR Y
        Number meanValue = dataset.getMeanValue(row, column);

+        if (meanValue == null) {
+            return; // nothing to do
+        }
        
        double value = meanValue.doubleValue();
@@ -465,15 +471,18 @@ protected void drawVerticalItem(Graphics2D g2,
 
        // standard deviation lines
-           double valueDelta = dataset.getStdDevValue(row, column).doubleValue();
-           double highVal = rangeAxis.valueToJava2D(meanValue.doubleValue()
-                   + valueDelta, dataArea, yAxisLocation);
-           double lowVal = rangeAxis.valueToJava2D(meanValue.doubleValue()
-                   - valueDelta, dataArea, yAxisLocation);
-
-           if (this.errorIndicatorStroke != null) {
-               g2.setStroke(this.errorIndicatorStroke);
-           }
-           else {
-               g2.setStroke(getItemOutlineStroke(row, column));
-           }
-           if (this.errorIndicatorPaint != null) {
-               g2.setPaint(this.errorIndicatorPaint);
-           }
-           else {
-               g2.setPaint(getItemOutlinePaint(row, column));
-           }
-           Line2D line = null;
-           line = new Line2D.Double(rectX + rectWidth / 2.0d, lowVal,
-                                    rectX + rectWidth / 2.0d, highVal);
-           g2.draw(line);
-           line = new Line2D.Double(rectX + rectWidth / 2.0d - 5.0d, highVal,
-                                    rectX + rectWidth / 2.0d + 5.0d, highVal);
-           g2.draw(line);
-           line = new Line2D.Double(rectX + rectWidth / 2.0d - 5.0d, lowVal,
-                                    rectX + rectWidth / 2.0d + 5.0d, lowVal);
-           g2.draw(line);
+           Number stdDev = dataset.getStdDevValue(row, column);
+           if (stdDev != null) {
+               double valueDelta = stdDev.doubleValue();
+               double highVal = rangeAxis.valueToJava2D(meanValue.doubleValue()
+                       + valueDelta, dataArea, yAxisLocation);
+               double lowVal = rangeAxis.valueToJava2D(meanValue.doubleValue()
+                       - valueDelta, dataArea, yAxisLocation);
+
+               if (this.errorIndicatorStroke != null) {
+                   g2.setStroke(this.errorIndicatorStroke);
+               }
+               else {
+                   g2.setStroke(getItemOutlineStroke(row, column));
+               }
+               if (this.errorIndicatorPaint != null) {
+                   g2.setPaint(this.errorIndicatorPaint);
+               }
+               else {
+                   g2.setPaint(getItemOutlinePaint(row, column));
+               }
+               Line2D line = null;
+               line = new Line2D.Double(rectX + rectWidth / 2.0d, lowVal,
+                                        rectX + rectWidth / 2.0d, highVal);
+               g2.draw(line);
+               line = new Line2D.Double(rectX + rectWidth / 2.0d - 5.0d, highVal,
+                                        rectX + rectWidth / 2.0d + 5.0d, highVal);
+               g2.draw(line);
+               line = new Line2D.Double(rectX + rectWidth / 2.0d - 5.0d, lowVal,
+                                        rectX + rectWidth / 2.0d + 5.0d, lowVal);
+               g2.draw(line);
+           }