diff --git a/src/java/org/apache/commons/lang/time/DurationFormatUtils.java b/src/java/org/apache/commons/lang/time/DurationFormatUtils.java
index 127b92795..0848d06b6 100644
--- a/src/java/org/apache/commons/lang/time/DurationFormatUtils.java
+++ b/src/java/org/apache/commons/lang/time/DurationFormatUtils.java
@@ -303,12 +303,14 @@ public class DurationFormatUtils {
             days -= 1;
         }
         while (days < 0) {
-            days += 31;
-//days += 31; // TODO: Need tests to show this is bad and the new code is good.
-// HEN: It's a tricky subject. Jan 15th to March 10th. If I count days-first it is 
-// 1 month and 26 days, but if I count month-first then it is 1 month and 23 days.
-// Also it's contextual - if asked for no M in the format then I should probably 
-// be doing no calculating here.
+            // Calculate the month we're borrowing from - it's the month we're currently at
+            Calendar monthOfBorrowing = Calendar.getInstance(timezone);
+            monthOfBorrowing.setTime(new Date(startMillis));
+            monthOfBorrowing.add(Calendar.YEAR, years);
+            monthOfBorrowing.add(Calendar.MONTH, months);
+
+            // Add the number of days in the month we're borrowing from
+            days += monthOfBorrowing.getActualMaximum(Calendar.DAY_OF_MONTH);
             months -= 1;
         }
         while (months < 0) {
