diff --git a/src/main/java/org/jsoup/nodes/Node.java b/src/main/java/org/jsoup/nodes/Node.java
index 15d8f2bd..e257f8cd 100644
--- a/src/main/java/org/jsoup/nodes/Node.java
+++ b/src/main/java/org/jsoup/nodes/Node.java
@@ -451,6 +451,7 @@ public abstract class Node implements Cloneable {
             nodes.add(child);
             child.setSiblingIndex(nodes.size()-1);
         }
+        nodelistChanged();
     }
 
     protected void addChildren(int index, Node... children) {
@@ -462,6 +463,7 @@ public abstract class Node implements Cloneable {
         }
         nodes.addAll(index, Arrays.asList(children));
         reindexChildren(index);
+        nodelistChanged();
     }
     
     protected void reparentChild(Node child) {
