diff --git a/src/main/java/org/apache/commons/lang3/RandomStringUtils.java b/src/main/java/org/apache/commons/lang3/RandomStringUtils.java
index 5c7ecae..655430f 100644
--- a/src/main/java/org/apache/commons/lang3/RandomStringUtils.java
+++ b/src/main/java/org/apache/commons/lang3/RandomStringUtils.java
@@ -230,6 +230,9 @@ public class RandomStringUtils {
             throw new IllegalArgumentException("Requested random string length " + count + " is less than 0.");
         }
 
+        if (chars != null && chars.length == 0) {
+            throw new IllegalArgumentException("The chars array must not be empty");
+        }
 
         if (start == 0 && end == 0) {
                 if (!letters && !numbers) {