--- a/src/org/mockito/internal/util/Timer.java
+++ b/src/org/mockito/internal/util/Timer.java
@@ -1,8 +1,11 @@
 package org.mockito.internal.util;
+import org.mockito.exceptions.misusing.FriendlyReminderException;
 
 public class Timer {
 
     private final long durationMillis;
     private long startTime = -1;
 
     public Timer(long durationMillis) {
+        if (durationMillis < 0) {
+            throw new FriendlyReminderException("Duration cannot be negative");
+        }
         this.durationMillis = durationMillis;
     }
 
     /**
      * Informs whether the timer is still counting down.
      */
     public boolean isCounting() {
         assert startTime != -1;
         return System.currentTimeMillis() - startTime <= durationMillis;
     }
 
     /**
      * Starts the timer count down.
      */
     public void start() {
         startTime = System.currentTimeMillis();
     }
 
 }