diff --git a/src/main/java/org/apache/commons/csv/CSVRecord.java b/src/main/java/org/apache/commons/csv/CSVRecord.java
index dc52017c..0b971ac9 100644
--- a/src/main/java/org/apache/commons/csv/CSVRecord.java
+++ b/src/main/java/org/apache/commons/csv/CSVRecord.java
@@ -82,7 +82,15 @@ public class CSVRecord implements Serializable, Iterable<String> {
                     "No header mapping was specified, the record values can't be accessed by name");
         }
         final Integer index = mapping.get(name);
-            return index != null ? values[index.intValue()] : null;
+        if (index != null) {
+            if (index.intValue() >= values.length || index.intValue() < 0) {
+                throw new IllegalArgumentException(
+                        String.format("'%s' has been mapped to an invalid index. The record has %d values, but the index is %d",
+                                name, values.length, index.intValue()));
+            }
+            return values[index.intValue()];
+        }
+        return null;
     }
 
     /**
