diff --git a/src/main/java/org/apache/commons/collections/list/SetUniqueList.java b/src/main/java/org/apache/commons/collections/list/SetUniqueList.java
index 57e54121..df881f56 100644
--- a/src/main/java/org/apache/commons/collections/list/SetUniqueList.java
+++ b/src/main/java/org/apache/commons/collections/list/SetUniqueList.java
@@ -233,8 +233,10 @@ public class SetUniqueList<E> extends AbstractSerializableListDecorator<E> {
             // (and it hasn't been swapped with itself)
             super.remove(pos); // remove the duplicate by index
         }
-        set.add(object); 
-        set.remove(removed);
+        set.add(object);
+        if (!object.equals(removed)) {
+            set.remove(removed);
+        }
         return removed; // return the item deleted by the set
     }
 
