diff --git a/src/java/org/apache/commons/cli/HelpFormatter.java b/src/java/org/apache/commons/cli/HelpFormatter.java
index 639b9d5..88aa9a6 100644
--- a/src/java/org/apache/commons/cli/HelpFormatter.java
+++ b/src/java/org/apache/commons/cli/HelpFormatter.java
@@ -868,11 +868,11 @@ public class HelpFormatter {
             return pos;
         }
         
-        // must look for the first whitespace chearacter after startPos 
+        // must look for the first whitespace chearacter after startPos
         // + width
         pos = startPos + width;
 
-        while ((pos <= text.length()) && ((c = text.charAt(pos)) != ' ')
+        while ((pos < text.length()) && ((c = text.charAt(pos)) != ' ')
                && (c != '\n') && (c != '\r'))
         {
             ++pos;
