diff --git a/src/org/mockito/internal/creation/DelegatingMethod.java b/src/org/mockito/internal/creation/DelegatingMethod.java
index 349719bd0..58c1f5c82 100644
--- a/src/org/mockito/internal/creation/DelegatingMethod.java
+++ b/src/org/mockito/internal/creation/DelegatingMethod.java
@@ -52,11 +52,21 @@ public class DelegatingMethod implements MockitoMethod {
      */
     @Override
     public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null) return false;
+
+        if (o instanceof DelegatingMethod) {
+            DelegatingMethod that = (DelegatingMethod) o;
+            return method.equals(that.method);
+        } else if (o instanceof Method) {
             return method.equals(o);
+        } else {
+            return false;
+        }
     }
 
     @Override
     public int hashCode() {
-        return 1;
+        return method.hashCode();
     }
 }