diff --git a/src/main/java/org/apache/commons/csv/CSVParser.java b/src/main/java/org/apache/commons/csv/CSVParser.java
index a6a6d499..17ba27fb 100644
--- a/src/main/java/org/apache/commons/csv/CSVParser.java
+++ b/src/main/java/org/apache/commons/csv/CSVParser.java
@@ -368,7 +368,11 @@ public final class CSVParser implements Iterable<CSVRecord>, Closeable {
             // build the name to index mappings
             if (header != null) {
                 for (int i = 0; i < header.length; i++) {
-                    hdrMap.put(header[i], Integer.valueOf(i));
+                    final String key = header[i];
+                    if (hdrMap.containsKey(key)) {
+                        throw new IllegalStateException("The header contains duplicate names: " + java.util.Arrays.toString(header));
+                    }
+                    hdrMap.put(key, Integer.valueOf(i));
                 }
             }
         }
