diff --git a/src/main/java/org/apache/commons/codec/language/bm/Lang.java b/src/main/java/org/apache/commons/codec/language/bm/Lang.java
index e9c2e728..b20a549f 100644
--- a/src/main/java/org/apache/commons/codec/language/bm/Lang.java
+++ b/src/main/java/org/apache/commons/codec/language/bm/Lang.java
@@ -134,7 +134,7 @@ public class Lang {
             throw new IllegalStateException("Unable to resolve required resource:" + LANGUAGE_RULES_RN);
         }
 
-        final Scanner scanner = new Scanner(lRulesIS, ResourceConstants.ENCODING);
+        final Scanner scanner = new Scanner(lRulesIS, ResourceConstants.ENCODING).useLocale(Locale.ROOT);
         try {
             boolean inExtendedComment = false;
             while (scanner.hasNextLine()) {
diff --git a/src/main/java/org/apache/commons/codec/language/bm/Languages.java b/src/main/java/org/apache/commons/codec/language/bm/Languages.java
index a791a2a6..54900819 100644
--- a/src/main/java/org/apache/commons/codec/language/bm/Languages.java
+++ b/src/main/java/org/apache/commons/codec/language/bm/Languages.java
@@ -21,6 +21,7 @@ import java.io.InputStream;
 import java.util.Collections;
 import java.util.EnumMap;
 import java.util.HashSet;
+import java.util.Locale;
 import java.util.Map;
 import java.util.NoSuchElementException;
 import java.util.Scanner;
@@ -175,7 +176,7 @@ public class Languages {
             throw new IllegalArgumentException("Unable to resolve required resource: " + languagesResourceName);
         }
 
-        final Scanner lsScanner = new Scanner(langIS, ResourceConstants.ENCODING);
+        final Scanner lsScanner = new Scanner(langIS, ResourceConstants.ENCODING).useLocale(Locale.ROOT);
         try {
             boolean inExtendedComment = false;
             while (lsScanner.hasNextLine()) {
diff --git a/src/main/java/org/apache/commons/codec/language/bm/Rule.java b/src/main/java/org/apache/commons/codec/language/bm/Rule.java
index f96ff7c4..90470d91 100644
--- a/src/main/java/org/apache/commons/codec/language/bm/Rule.java
+++ b/src/main/java/org/apache/commons/codec/language/bm/Rule.java
@@ -30,6 +30,7 @@ import java.util.Map;
 import java.util.Scanner;
 import java.util.Set;
 import java.util.regex.Matcher;
+import java.util.Locale;
 import java.util.regex.Pattern;
 
 import org.apache.commons.codec.language.bm.Languages.LanguageSet;
@@ -251,7 +252,7 @@ public class Rule {
             throw new IllegalArgumentException("Unable to load resource: " + resName);
         }
 
-        return new Scanner(rulesIS, ResourceConstants.ENCODING);
+        return new Scanner(rulesIS, ResourceConstants.ENCODING).useLocale(Locale.ROOT);
     }
 
     private static Scanner createScanner(final String lang) {
@@ -262,7 +263,7 @@ public class Rule {
             throw new IllegalArgumentException("Unable to load resource: " + resName);
         }
 
-        return new Scanner(rulesIS, ResourceConstants.ENCODING);
+        return new Scanner(rulesIS, ResourceConstants.ENCODING).useLocale(Locale.ROOT);
     }
 
     private static boolean endsWith(final CharSequence input, final CharSequence suffix) {
