--- a/source/org/jfree/chart/plot/CategoryPlot.java
+++ b/source/org/jfree/chart/plot/CategoryPlot.java
@@ -2162,6 +2162,9 @@
         else {
             markers = (ArrayList) this.backgroundDomainMarkers.get(new Integer(
                     index));
         }
+        if (markers == null) {
+            return false;
+        }
         boolean removed = markers.remove(marker);
         if (removed && notify) {
             fireChangeEvent();
@@ -2444,6 +2444,9 @@
         else {
             markers = (ArrayList) this.backgroundRangeMarkers.get(new Integer(
                     index));
         }
+        if (markers == null) {
+            return false;
+        }
         boolean removed = markers.remove(marker);
         if (removed && notify) {
             fireChangeEvent();
         }