diff --git a/gson/src/main/java/com/google/gson/internal/UnsafeAllocator.java b/gson/src/main/java/com/google/gson/internal/UnsafeAllocator.java
index 49fdf7f6..60798d6a 100644
--- a/gson/src/main/java/com/google/gson/internal/UnsafeAllocator.java
+++ b/gson/src/main/java/com/google/gson/internal/UnsafeAllocator.java
@@ -45,6 +45,9 @@ public abstract class UnsafeAllocator {
         @Override
         @SuppressWarnings("unchecked")
         public <T> T newInstance(Class<T> c) throws Exception {
+          if (c.isInterface() || java.lang.reflect.Modifier.isAbstract(c.getModifiers())) {
+            throw new UnsupportedOperationException("Cannot allocate " + c);
+          }
           return (T) allocateInstance.invoke(unsafe, c);
         }
       };
@@ -68,6 +71,9 @@ public abstract class UnsafeAllocator {
         @Override
         @SuppressWarnings("unchecked")
         public <T> T newInstance(Class<T> c) throws Exception {
+          if (c.isInterface() || java.lang.reflect.Modifier.isAbstract(c.getModifiers())) {
+            throw new UnsupportedOperationException("Cannot allocate " + c);
+          }
           return (T) newInstance.invoke(null, c, constructorId);
         }
       };
@@ -87,6 +93,9 @@ public abstract class UnsafeAllocator {
         @Override
         @SuppressWarnings("unchecked")
         public <T> T newInstance(Class<T> c) throws Exception {
+          if (c.isInterface() || java.lang.reflect.Modifier.isAbstract(c.getModifiers())) {
+            throw new UnsupportedOperationException("Cannot allocate " + c);
+          }
           return (T) newInstance.invoke(null, c, Object.class);
         }
       };
