diff --git a/src/java/org/apache/commons/collections/keyvalue/MultiKey.java b/src/java/org/apache/commons/collections/keyvalue/MultiKey.java
index 7aaa7ecf..2939c2ec 100644
--- a/src/java/org/apache/commons/collections/keyvalue/MultiKey.java
+++ b/src/java/org/apache/commons/collections/keyvalue/MultiKey.java
@@ -269,7 +269,11 @@ public class MultiKey implements Serializable {
 	/**
 	 * Recalculate the hash code after deserialization. The hash code of some
 	 * keys might have change (hash codes based on the system hash code are
-	 * only stable for the same process). 
+	 * only stable for the same process).
 	 * @return the instance with recalculated hash code
 	 */
+	private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
+		in.defaultReadObject();
+		calculateHashCode(keys);
+	}
 }
