diff --git a/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java b/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java
index bb670f30..e75b3992 100644
--- a/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java
+++ b/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java
@@ -195,7 +195,11 @@ public class TarArchiveInputStream extends ArchiveInputStream {
             return null;
         }
 
-        currEntry = new TarArchiveEntry(headerBuf);
+        try {
+            currEntry = new TarArchiveEntry(headerBuf);
+        } catch (IllegalArgumentException e) {
+            throw new IOException("Error detected parsing the header", e);
+        }
         entryOffset = 0;
         entrySize = currEntry.getSize();
 
