diff --git a/src/main/java/org/jsoup/nodes/Entities.java b/src/main/java/org/jsoup/nodes/Entities.java
index 08cf3113..8775281d 100644
--- a/src/main/java/org/jsoup/nodes/Entities.java
+++ b/src/main/java/org/jsoup/nodes/Entities.java
@@ -118,9 +118,11 @@ public class Entities {
                             accum.append("&#xa0;");
                         break;
                     case '<':
-                        // escape when in character data or when in a xml attribue val; not needed in html attr val
+                        // escape when in character data or when in a xml attribute val; not needed in html attr val
                         if (!inAttribute)
                             accum.append("&lt;");
+                        else if (escapeMode == EscapeMode.xhtml)
+                            accum.append("&lt;");
                         else
                             accum.append(c);
                         break;
