diff --git a/src/main/java/org/jsoup/helper/HttpConnection.java b/src/main/java/org/jsoup/helper/HttpConnection.java
index f8fd9dd7..f8deb6aa 100644
--- a/src/main/java/org/jsoup/helper/HttpConnection.java
+++ b/src/main/java/org/jsoup/helper/HttpConnection.java
@@ -770,8 +770,15 @@ public class HttpConnection implements Connection {
                             cookie(cookieName, cookieVal);
                     }
                 } else { // combine same header names with comma: http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
-                    if (!values.isEmpty())
-                        header(name, values.get(0));
+                    if (!values.isEmpty()) {
+                        StringBuilder headerBuf = new StringBuilder();
+                        for (int i = 0; i < values.size(); i++) {
+                            if (i > 0)
+                                headerBuf.append(", ");
+                            headerBuf.append(values.get(i));
+                        }
+                        header(name, headerBuf.toString());
+                    }
                 }
             }
         }
